FORMS.md

```diff
@@ -0,0 +1,243 @@
+<!-- This file will not be merged. -->
+
+# Form Components
+
+The purpose of this document is to outline the spectrum of components required to make up a complete Form. This includes: Input, Field, FieldGroup, and Form.
+
+Along with a general description and suggested structure, each component below includes a list of possible attributes and states. Design should review these lists to ensure that any values that require user affordances are accounted for. For example: `disabled` or `:invalid`.
+
+**Note:** All code examples are for illustrative purposes only. Structure and behavior may change based on subsequent designs for these components. There are also opportunities for re-use across components; for example, Form.Title extending from Title.
+
+## Inputs
+
+Inputs are the atomic units around which the following components are built. Inputs represent unique controls and are not covered in this document, which seeks to outline the ubiquitous aspects of Forms.
+
+For the sake of clarity, when an Input component is referenced below, it includes only the concerns that make it unique. For example, an Input of the type RadioButtonGroup includes a set of RadioButtons, but *not* their associated Label, Hint, or Error. Those components belong to Field.
+
+## Field
+
+Field is the primary component for assembling forms. It includes all of the supporting context required by the Input. This component would also be responsible for managing the appropriate application of `aria-` attributes and `id` associations between Inputs and their associated elements.
+
+Because the requirements for both behavior and display of fields are stringent, I recommend that authors not rely on composition to assemble their fields if possible.
+
+### Non-global attrs
+
+#### Fieldset-specific HTML attributes
+
+- `disabled`: If this Boolean attribute is set, all Inputs that are descendants of the `<fieldset>` are disabled.
+  - In the case of Field, putting this on the `fieldset` would have the desired behavior, but passing `disabled` to the Input itself would be less brittle.
+- `name`: The name associated with the group.
+- `form`: This attribute takes the value of the id attribute of a `<form>` element you want the `<fieldset>` to be part of, even if it is not inside the form. Please note that usage of this is confusing — if you want the `<input>` elements inside the `<fieldset>` to be associated with the form, you need to use the form attribute directly on those elements.
+  - In the case of Field, this attribute should be passed to Input instead of the containing `fieldset` to create the ideal behavior here.
```

Comment:
I'm not sure I follow this. Are we trying to say that our approach is to encourage composition and have consumers place their input components as descendants of a parent form component so that they are associated with the form correctly?

Reply:
These attribute sections speak solely to the available HTML attributes that an author might use or expect to use when assembling their form/field. Due diligence hasn't always been done in the past, which has led to undesirable behaviors becoming standard implementations.

This note about `form` speaks specifically to the unexpected behavior of placing the attribute on a `fieldset` vs an `input`. However, since we won't be using `fieldset` as the container for `Field`, this note becomes somewhat moot.